germsvel/phoenix_test
# Add Wallaby driver

Fixes #73 

## Done
- add Wallaby driver
- prepare test setup for Wallaby (add app.js, convert `Endpoint` to proper/regular endpoint)
- move `Assertions` into `Driver` protocol (wrap with retry for Wallaby)
- run existing tests twice via `test_also_with_js` macro

## To do
- documentation
  - update main moduledoc: third Driver `Wallaby`, use via `@tag :js`
  - caveat: `conn` passed to `visit/2` is mostly ignored (not supported: `conn.assigns`, supported: `conn.req_cookies`)
  - possible discrepancies:
    - errors for fields without an ID (if no unique selector can be generated from other attributes) 
- fix `NoRouteError` tests (broken due to changes to `Endpoint`)
- `open_browser`
  - fill in form values (not automatically preserved in DOM tree via `value/selected/checked` attributes)
  - check stylesheets are loaded correctly
- copy initial `conn.req_cookies` via `Wallaby.Browser.set_cookie/4`
